packages/opentelemetry/src/utils/mapStatus.ts

```diff
@@ -48,28 +33,40 @@ export function mapStatus(span: AbstractSpan): SpanStatus {
   const attributes = spanHasAttributes(span) ? span.attributes : {};
   const status = spanHasStatus(span) ? span.status : undefined;
 
-  const httpCode = attributes[SemanticAttributes.HTTP_STATUS_CODE];
-  const grpcCode = attributes[SemanticAttributes.RPC_GRPC_STATUS_CODE];
-
-  const code = typeof httpCode === 'string' ? httpCode : typeof httpCode === 'number' ? httpCode.toString() : undefined;
-  if (code) {
-    const sentryStatus = canonicalCodesHTTPMap[code];
-    if (sentryStatus) {
-      return { code: SPAN_STATUS_ERROR, message: sentryStatus };
+  if (status) {
+    // Since span status OK is not set by default, we give it priority: https://opentelemetry.io/docs/concepts/signals/traces/#span-status
+    if (status.code === SpanStatusCode.OK) {
+      return { code: SPAN_STATUS_OK };
+      // If the span is already marked as erroneous we return that exact status
+    } else if (status.code === SpanStatusCode.ERROR) {
+      return { code: SPAN_STATUS_ERROR, message: status.message };
     }
   }
 
-  if (typeof grpcCode === 'string') {
-    const sentryStatus = canonicalCodesGrpcMap[grpcCode];
-    if (sentryStatus) {
-      return { code: SPAN_STATUS_ERROR, message: sentryStatus };
-    }
+  // If the span status is UNSET, we try to infer it from HTTP or GRPC status codes.
+
+  const httpCodeAttribute = attributes[SemanticAttributes.HTTP_STATUS_CODE];
+  const grpcCodeAttribute = attributes[SemanticAttributes.RPC_GRPC_STATUS_CODE];
+
+  const numberHttpCode =
+    typeof httpCodeAttribute === 'number'
+      ? httpCodeAttribute
+      : typeof httpCodeAttribute === 'string'
+        ? parseInt(httpCodeAttribute)
+        : undefined;
+
+  if (numberHttpCode) {
+    return getSpanStatusFromHttpCode(numberHttpCode);
   }
 
-  const statusCode = status && status.code;
-  if (statusCode === SpanStatusCode.OK || statusCode === SpanStatusCode.UNSET) {
-    return { code: SPAN_STATUS_OK };
+  if (typeof grpcCodeAttribute === 'string') {
```

Comment:
m: can `grpcCodeAttribute` be of any other type than `string` like `httpCodeAttribute`?

Reply:
I opened an issue to track this, as I didn't touch that particular logic: https://github.com/getsentry/sentry-javascript/issues/11195

It may very well be that this is a leftover from our first initial OTEL implementation draft.